### PR TITLE
Move quoted.{Expr|Type}.apply to scala.internal

### PR DESCRIPTION
**Pull request by nicolasstucki**
This is to avoid having this internal representation leaked to the user with `scala.quoted.Expr.apply`. Users could have accidentally quoted a term using `Expr(myTerm)` which does not get the correct level in the context while typing.

### CODE REVIEW

File: library/src-bootstrapped/scala/compiletime/Type.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package scala.compiletime.quoted
+
+import scala.quoted._
+
+object Type {
+
+  /** A term quote is desugared by the compiler into a call to this method */
```
Comment (liufengyun) [inline library/src-bootstrapped/scala/compiletime/Type.scala]: Typo: `term quote` -> `type quote`

File: library/src-non-bootstrapped/scala/compiletime/quoted/Type.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package scala.compiletime.quoted
+
+import scala.quoted._
+
+object Type {
+
+  /** A term quote is desugared by the compiler into a call to this method */
+  def apply[T]: Type[T] =
+    throw new Error("Internal error: this method call should have been replaced by the compiler")
```
Comment (liufengyun) [inline library/src-non-bootstrapped/scala/compiletime/quoted/Type.scala]: Typo: `term quote` -> `type quote`

File: library/src-bootstrapped/scala/compiletime/Type.scala (hunk 1 above)
Comment (nicolasstucki) [reply library/src-bootstrapped/scala/compiletime/Type.scala]: fixed

File: library/src-non-bootstrapped/scala/compiletime/quoted/Type.scala (hunk 2 above)
Comment (nicolasstucki) [reply library/src-non-bootstrapped/scala/compiletime/quoted/Type.scala]: fixed

**Comment by nicolasstucki** [comment]
If we only changed the method name but kept it in `scala.quoted.Expr` it would be exposed when completing `Expr.` in the IDE or REPL.

**Comment by nicolasstucki** [comment]
Actually, the `compiletime` package is not the best place to put it.

**Comment by nicolasstucki** [comment]
@liufengyun I moved thing around

**Comment by odersky** [comment]
Let's discuss the name. 

We have `scala.compiletime` as user-facing the inline support package, and `scala.internal` for things that only the compiler should know about. So I think this should in a package object in `scala.internal`. If we must use `Compiletime` it should be in lower case.

**Comment by nicolasstucki** [comment]
Then it should be in `scala.internal.Expr` or `scala.internal.quoted.Expr`

**Comment by nicolasstucki** [comment]
@liufengyun now the logic was moved to `scala.internal` which should finally be the correct place to put it in.

**Pull request merged**